### PR TITLE
Cleanup reification of primitive liftables

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/transform/ReifyQuotes.scala [hunk 1]
```diff
@@ -206,24 +206,19 @@ class ReifyQuotes extends MacroTransform {
         qctx
       }
 
-      def liftedValue[T](value: T, name: TermName) =
-        ref(defn.LiftableModule)
-          .select(name).appliedToType(originalTp)
-          .select("toExpr".toTermName).appliedTo(Literal(Constant(value)))
+      def liftedValue[T](const: Constant) = {
```
Comment (smarter) [inline compiler/src/dotty/tools/dotc/transform/ReifyQuotes.scala]: unused type parameter

File: compiler/src/dotty/tools/dotc/transform/ReifyQuotes.scala [hunk 2]
```diff
@@ -206,24 +206,19 @@ class ReifyQuotes extends MacroTransform {
...
+        val ltp = defn.LiftableClass.typeRef.appliedTo(ConstantType(const))
+        val liftable = ctx.typer.inferImplicitArg(ltp, body.span)
+        if (liftable.tpe.isInstanceOf[SearchFailureType])
+          ctx.error(ctx.typer.missingArgMsg(liftable, ltp, "Could no optimize constant in quote"), ctx.source.atSpan(body.span))
+        liftable.select("toExpr".toTermName).appliedTo(Literal(const))
```
Comment (smarter) [inline compiler/src/dotty/tools/dotc/transform/ReifyQuotes.scala]: toExpr should be added to StdNames

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/transform/ReifyQuotes.scala]: Done

**Comment by nicolasstucki** [comment]
It was mostly to be able to rename them easily. I will remove the implicit search.

**Pull request merged**